### PR TITLE
ocrd-tool: allow definition of input/output file groups

### PR DESCRIPTION
**Pull request by kba**
@bertsky

### CODE REVIEW

**Comment by bertsky** [comment]
Splendid!

File: ocrd_tool.md [hunk 1]
```diff
@@ -59,6 +73,16 @@ properties:
           executable:
             description: The name of the CLI executable in $PATH
             type: string
+          input_file_grp:
+            description: Input fileGrp@USE this tool expects by default
+            type: array
+            items:
+              type: string
+              pattern: '^OCR-D-[A-Z0-9-]+$'
```
Comment (VolkerHartmann) [inline ocrd_tool.md]: Typo?: [A-Z0-9\\-]

File: ocrd_tool.md [hunk 2]
```diff
@@ -59,6 +73,16 @@ properties:
...
+          output_file_grp:
+            description: Output fileGrp@USE this tool produces by default
+            type: string
+            pattern: '^OCR-D-[A-Z0-9-]+$'
```
Comment (VolkerHartmann) [inline ocrd_tool.md]: Typo?: [A-Z0-9\\-]
The backslash inside the square bracket has to be escaped by an additional backslash.

File: ocrd_tool.md (hunk 1 above)
Comment (kba) [reply ocrd_tool.md]: dash at the end of a character group is not interpreted as a meta char, c.f. https://regex101.com/r/4d8P1b/1.

**Comment by kba** [comment]
Should probably be even required, otherwise nobody will use it and workflow engine cannot rely on it.

File: ocrd_tool.md (hunk 1 above)
Comment (VolkerHartmann) [reply ocrd_tool.md]: Ok, I didn't know that yet.

**Pull request merged**